source/common/ur_singleton.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#pragma once
+#include <memory>
+#include <unordered_map>
+#include <mutex>
+
+//////////////////////////////////////////////////////////////////////////
+/// a abstract factory for creation of singleton objects
+template<typename _singleton_t, typename _key_t>
```

Comment:
I think prefixing types with `_` is UB since they are reserved by the compiler.

Reply:
Done